File: lab-2/src/moves/EnergyBall.java
package moves;

import ru.ifmo.se.pokemon.*;
public class EnergyBall extends SpecialMove{
    public EnergyBall(){
        super(Type.GRASS, 90, 100);
    }

    protected void applyOppEffects(Pokemon p){
        if(Math.random()<0.1D) {
            p.setMod(Stat.SPECIAL_DEFENSE, -1);
        }
    }

    public String describe(){
        return "is using Energy Ball";
    }

}

File: lab-2-kotlin/src/moves/EnergyBall.kt
package moves

import ru.ifmo.se.pokemon.*

class EnergyBall : SpecialMove(Type.GRASS, 90.0, 100.0) {
    override fun applyOppEffects(p: Pokemon) {
        if (Math.random() < 0.1) {
            p.setMod(Stat.SPECIAL_DEFENSE, -1)
        }
    }

    public override fun describe(): String {
        return "is using Energy Ball"
    }
}